internal/app/machined/pkg/controllers/cluster/local_affiliate.go

```diff
@@ -202,6 +203,11 @@ func (ctrl *LocalAffiliateController) Run(ctx context.Context, r controller.Runt
 							continue
 						}
 
+						if slices.Contains(spec.Addresses, ip) {
```

Comment:
NodeAddresses shouldn't contain duplicates in the first place. if it does, let's better fix `IPs()` method on the `NodeAddress` spec resource (see line 196 above)